src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/common/WithHistorySupportFsm.java

```diff
@@ -99,24 +122,41 @@ public void saveNewEventToHistory(String action, FlowEventData.Event event) {
         saveNewEventToHistory(action, event, null, null);
     }
 
+    /**
+     * Add a history record on the new event.
+     */
+    public void saveNewEventToHistory(String flowId, String action, FlowEventData.Event event) {
+        String taskId = KeyProvider.joinKeys(flowId, getCommandContext().getCorrelationId());
+        saveNewEventToHistory(flowId, action, event, null, null, taskId);
+    }
+
     /**
      * Add a history record on the new event.
      */
     public void saveNewEventToHistory(String action, FlowEventData.Event event,
                                       FlowEventData.Initiator initiator,
                                       String details) {
-        log.debug("Flow {} action - {} : {}", getFlowId(), action, event);
+        saveNewEventToHistory(getFlowId(), action, event, initiator, details, getCommandContext().getCorrelationId());
+    }
+
+    /**
+     * Add a history record on the new event.
+     */
+    public void saveNewEventToHistory(String flowId, String action, FlowEventData.Event event,
```

Comment:
So now, a FSM can log an history event with the taskId and flowId which don't belong to the FSM? Let's discuss this.